SpatialGDK/Documentation/content/singleton-actors.md

```diff
@@ -1,30 +1,25 @@
-
 # Singleton Actors
 
 Singleton Actors allow a single source of truth for both operations and data across a multi-server simulation. They are server-side authoritative [Actors](https://docs.unrealengine.com/en-us/Programming/UnrealArchitecture/Actors) that are restricted to one instantiation on SpatialOS.
 
 There are two kinds of Singleton Actors:
 
 * **Public Singleton Actors** - Singleton Actors which are replicated to server-workers and client-workers. `GameState` is a Public Singleton Actor.
-* **Private Singletons** - Singleton Actors which are replicated to server-workers, but not accessible to client-workers. `GameMode` is a Private Singleton Actor.
+* **Private Singleton Actors** - Singleton Actors which are replicated to server-workers, but not accessible to client-workers. `GameMode` is a Private Singleton Actor.
 
-You can define any class as a Singleton Actor. At the moment the GDK for Unreal only supports Public Singleton Actors.
+You can define any class as a Singleton Actor.
 
-Each server-worker should instantiate their own local version of each Singleton Actor. For `GameMode` and `GameState`, Unreal Engine does this automatically.
+Each server-worker should instantiate their own local version of each Singleton Actor. For `GameMode` and `GameState`, Unreal Engine does this automatically. Client-workers receive public singletons from the server-workers via the normal Actor replication lifecycle.
 
-Due to Unreal server-workers spawning their own instances of each Singleton Actor, proper replication and authority management of Singleton Actors becomes a bit tricky. To solve this issue, we have introduced the concept of a Global State Manager (GSM) to enable proper replication of Singleton Actors. The GSM solves the problem of replicating Singleton Actors by only allowing the server-worker with [authority](https://docs.improbable.io/reference/latest/shared/glossary#read-and-write-access-authority) over the GSM to execute the initial replication of these Actors. All other server-workers will then link their local Singleton Actors to their respective SpatialOS entity.
+Due to Unreal server-workers spawning their own instances of each Singleton Actor, proper replication and authority management of Singleton Actors becomes a bit tricky. To solve this issue, we have introduced the concept of a Global State Manager (GSM) to enable proper replication of Singleton Actors. The GSM solves the problem of replicating Singleton Actors by only allowing the server-worker with [authority](https://docs.improbable.io/reference/latest/shared/glossary#read-and-write-access-authority) over the GSM to execute the initial replication of these Actors. All other server-workers will then link their local Singleton Actors to their respective SpatialOS entity. Because of this, you must update your snapshot whenever adding a new Singleton Actor to your project.
```

Comment:
In the context of the other new docs, this is now a bit confusing...
"Due to Unreal server-workers spawning their own instances of each Singleton Actor,". 

Do we mean native Unreal networking servers or do we mean SpatialOS server-workers?

---
Glossary link for authority:
[authority]({{urlroot}}/content/glossary#authority)

Reply:
Link to our glossary authority definition, instead of the improbable one?

Reply:
re: your first comment, both? In native unreal, the single server-worker, would create all the singletons. In Spatial Unreal, each server-worker creates the singletons, so we need additional logic to link them together, and ensure that only one of the server-workers is actually authoritative over them.

Reply:
RE: Glossary - 
Yes, keep peeps in the "padded cell" of Unreal land. The Glossary links out to the SpatialOS docs - and has a note that you are entering SpatialOS land where things might be a bit different as the content assumes you are using the SpatialOS (worker) SDK.